tests/jballerina-unit-test/src/test/resources/test-src/statements/matchstmt/const-pattern.bal

```diff
@@ -688,6 +688,68 @@ function constPatternWithPredeclaredPrefix2(int[] x) returns int {
     }
 }
 
+function testMatchStmtInsideForeachString1() {
+    string input = "1x2";
+    string output = "";
+    foreach string char in input {
+        match char {
+            "1" => {
+                output = output.concat("One");
+            }
+            "2" => {
+                output = output.concat("Two");
+            }
+            _ => {
+                output = output.concat("Other");
+            }
+        }
+    }
+    assertEquals("OneOtherTwo", output);
+}
+
+type CHAR string;
+
+function testMatchStmtInsideForeachString2() {
+    CHAR input = "1x2";
+    string output = "";
+    foreach string char in input {
+        match char {
+            "1" => {
+                output = output.concat("One");
+            }
+            "2" => {
+                output = output.concat("Two");
+            }
+            _ => {
+                output = output.concat("Other");
+            }
+        }
+    }
+    assertEquals("OneOtherTwo", output);
+}
+
+function testMatchStmtInsideForeachInt() {
+    int:Signed8[] input = [7, 2, 2];
+    string output = "";
+    foreach int number in input {
+        match number {
+            2 => {
+                output = output.concat("Two");
```

Comment:
Isn't it better if we use `output += "Two";`.
Using `concat` means we depend on `langlib string` module for this test.

Reply:
`langlib string` is a builtin module we can use. we can depend on that. I don't see a any specific reason not to